plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/serdes/IcebergFileScanTaskWrapper.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.serdes;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.apache.iceberg.FileScanTask;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.iceberg.util.SerializationUtil.deserializeFromBase64;
+import static org.apache.iceberg.util.SerializationUtil.serializeToBase64;
+
+public class IcebergFileScanTaskWrapper
```

Comment:
As we discussed, Iceberg is adding full Jackson support for its object models. This wrapper will be updated to use Jackson serialization once that is completed. Meanwhile I delegate everything to Java serialization that Iceberg provides. Because of that, some related classes are added `Serializable` or `Externalizable` implementations.

Reply:
What is the timeline for that. Will it be feasible to wait for Jackson serialization before merging this PR?

Reply:
yeah it's up to the community, I am just posting this out for review first while working on that. The 0.13 train is going to be out soon so we will target 0.14 for this support, so likely around Jan 2022.

Reply:
Java serialization is a hard no for Trino.  It is huge ugly mess with lots of security problems.  Instead, for temporary storage you could do something like annotate the iceberg objects with Jackson annotations or use Jackson features to [serialize third party objects](https://web.archive.org/web/20110829212918/http://wiki.fasterxml.com/JacksonMixInAnnotations).  For long term storage, you should write your own serialization objects and copy in and out of them.